trustmanager/remoteks/client.go

```diff
@@ -0,0 +1,94 @@
+package remoteks
+
+import (
+	"crypto/tls"
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/trustmanager"
+	google_protobuf "github.com/golang/protobuf/ptypes/empty"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// RemoteStore is a wrapper around the GRPC storage client, translating between
+// the Go and GRPC APIs.
+type RemoteStore struct {
+	client   StoreClient
+	location string
+}
+
+var _ trustmanager.Storage = &RemoteStore{}
+
+// NewRemoteStore instantiates a RemoteStore.
+func NewRemoteStore(server string, tlsConfig *tls.Config) (*RemoteStore, error) {
+	cc, err := grpc.Dial(
+		server,
+		grpc.WithTransportCredentials(
+			credentials.NewTLS(tlsConfig),
+		),
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &RemoteStore{
+		client:   NewStoreClient(cc),
+		location: server,
+	}, nil
+}
+
+// Set stores the data using the provided fileName
+func (s *RemoteStore) Set(fileName string, data []byte) error {
+	sm := &SetMsg{
+		FileName: fileName,
+		Data:     data,
+	}
+	_, err := s.client.Set(context.Background(), sm)
+	return err
+}
+
+// Remove deletes a file from the store relative to the store's base directory.
+// The path is cleaned before being made absolute to ensure no path traversal
+// outside the base directory is possible.
+func (s *RemoteStore) Remove(fileName string) error {
+	fm := &FileNameMsg{
+		FileName: fileName,
+	}
+	_, err := s.client.Remove(context.Background(), fm)
+	return err
+}
+
+// Get returns the file content found at fileName relative to the base directory
+// of the file store. The path is cleaned before being made absolute to ensure
```

Comment:
GH is unfortunately collapsing many diffs and so I'm having trouble finding the cleaning logic - could you please point me to it?